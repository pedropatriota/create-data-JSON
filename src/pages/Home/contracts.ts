import type { IFormFactory, Item } from "../../utils/contracts";

export type TValue =
  | (
      | Item
      | {
          latitude: string | number;
          longitude: string | number;
        }
      | {
          download: string | number;
          upload: string | number;
        }
    )[]
  | undefined;

export type TInputValue =
  | (TValue & string)
  | number
  | readonly string[]
  | undefined;

export interface IFormFields {
  formItem: IFormFactory;
  index: number;
  fieldName: string;
  value: TValue;
  inputValue: TInputValue;
  handleChange: (
    e: React.ChangeEvent<HTMLInputElement>,
    formItem: IFormFactory
  ) => void;
  handleSelectChange: (newValue: unknown, fieldName: string) => void;
}
