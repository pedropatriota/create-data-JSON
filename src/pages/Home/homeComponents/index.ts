import FormFields from "./Form";

export { FormFields };
