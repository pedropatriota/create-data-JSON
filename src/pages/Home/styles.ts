import styled from "styled-components";

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 3rem;
  background: #f8fbff;

  pre {
    margin: 3rem;
    font-size: 1.5rem;
  }
`;

export const FormContainer = styled.form`
  width: 100%;
  background: #fff;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  gap: 1.2rem;
`;
