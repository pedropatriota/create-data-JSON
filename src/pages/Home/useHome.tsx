import { FormEvent, useState } from "react";
import { initialFormData } from "../../utils";
import type { IFormData, IFormFactory, Item } from "../../utils/contracts";

const useHome = () => {
  const [formData, setFormData] = useState<IFormData>(initialFormData);
  const [dataJSON, setDataJSON] = useState<string | null>(null);

  const handleChange = (
    e: React.ChangeEvent<HTMLInputElement>,
    formItem: IFormFactory
  ) => {
    const { name, value } = e.target;
    if (!/^\d*(\.\d*)?$/.test(value)) {
      return;
    }
    const fieldName = formItem?.name;
    const fieldsLength = formItem?.fields?.length as number;
    const index = fieldsLength / fieldsLength - 1;

    setFormData((prevFormData) => {
      if (Array.isArray(formItem?.fields) && !!fieldName) {
        const updated = prevFormData[fieldName as keyof IFormData];

        updated[index] = { ...updated[index], [name]: value };

        return {
          ...prevFormData,
          [fieldName]: updated,
        };
      } else {
        return prevFormData;
      }
    });
  };

  const handleSelectChange = (newValue: unknown, field: string) => {
    setFormData({
      ...formData,
      [field]: newValue,
    });
  };

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();

    const addresses = formData?.addresses.map(({ latitude, longitude }) => {
      return {
        coordinates: {
          latitude: parseFloat(latitude as string),
          longitude: parseFloat(longitude as string),
        },
      };
    });

    const jsonData = {
      trainingData: [
        {
          addresses: [...addresses],
          bandwidths: [...formData.bandWidth]?.map((item) => ({
            download: parseFloat(item.download as string),
            upload: parseFloat(item.upload as string),
          })),
          distanceClasses: [...formData.distanceClasses]?.map(
            (item) => item.value
          ),
          technologyTypes: [...formData.technologyTypes]?.map(
            (item) => item.value
          ),
          connectionTypes: [...formData.connectionTypes]?.map(
            (item) => item.value
          ),
          networks: [...formData.networks]?.map((item) => item.value),
          contractTerms: [...formData.contractTerms]?.map((item) => item.value),
          areaCodes: [...formData.areaCodes]?.map((item) => item.value),
        },
      ],
    };

    setDataJSON(JSON.stringify(jsonData, null, 2));
    setFormData(initialFormData);
  };

  const getFieldValue = (formItem: IFormFactory, index: number) => {
    if (formItem?.fields?.length) {
      const fieldName = formItem?.fields?.map((field) => field.name)[index];
      const value = formData?.[fieldName as keyof IFormData];

      return value;
    }
    const fieldName = formItem?.name as string;
    const values = formData?.[fieldName as keyof IFormData];

    return values?.map((value) => value);
  };

  return {
    handleSubmit,
    formData,
    handleSelectChange,
    handleChange,
    getFieldValue,
    dataJSON,
  };
};

export default useHome;
