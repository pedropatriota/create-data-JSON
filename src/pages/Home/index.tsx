import { formFactory } from "../../utils";
import { FormFields } from "./homeComponents";
import useHome from "./useHome";
import * as Styled from "./styles";
import { TInputValue, TValue } from "./contracts";
import { Button } from "../../components";

const Home = () => {
  const {
    handleChange,
    handleSelectChange,
    handleSubmit,
    getFieldValue,
    dataJSON,
  } = useHome();

  return (
    <Styled.Container>
      <Styled.FormContainer onSubmit={handleSubmit}>
        {formFactory.map((formItem, index) => (
          <FormFields
            key={index}
            index={index}
            formItem={formItem}
            value={getFieldValue(formItem, index) as TValue}
            inputValue={getFieldValue(formItem, index) as TInputValue}
            handleSelectChange={handleSelectChange}
            handleChange={handleChange}
            fieldName={
              formItem?.fields
                ? formItem?.fields[index]?.name
                : (formItem?.name as string)
            }
          />
        ))}
        <Button type="submit">Generate</Button>
      </Styled.FormContainer>
      {dataJSON && <pre data-testid="dataJSON">{dataJSON}</pre>}
    </Styled.Container>
  );
};

export default Home;
