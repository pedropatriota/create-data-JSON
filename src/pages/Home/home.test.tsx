import {
  screen,
  render,
  fireEvent,
  getByPlaceholderText,
} from "@testing-library/react";

import { userEvent } from "@testing-library/user-event";
import Home from ".";

let mockHandleSubmit = jest.fn();
let mockHandleSelectChange = jest.fn();
let mockHandleChange = jest.fn();

jest.mock("./useHome", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    handleSubmit: mockHandleSubmit,
    handleSelectChange: mockHandleSelectChange,
    handleChange: mockHandleChange,
    getFieldValue: jest.fn(),
  })),
}));

describe("test Home page", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render the initial state", () => {
    render(<Home />);

    const inputs = screen.getAllByRole("textbox");
    const selects = screen.getAllByRole("combobox");
    const button = screen.getByRole("button");

    expect(selects).toHaveLength(6);
    expect(inputs).toHaveLength(4);
    expect(button).toBeInTheDocument();
  });

  it("should fail when a required input or select is not filled", async () => {
    render(<Home />);

    const button = screen.getByRole("button");
    const dataJSON = screen.queryByTestId("dataJSON");

    expect(dataJSON).not.toBeInTheDocument();

    await userEvent.click(button);

    expect(mockHandleSubmit).not.toHaveBeenCalled();
  });

  it("should submit when the required inputs/selects are filled", async () => {
    render(<Home />);

    const button = screen.getByRole("button");
    const dataJSON = screen.queryByTestId("dataJSON");

    const latitudeInput = screen.getByPlaceholderText(/latitude/i);
    const longitudeInput = screen.getByPlaceholderText(/longitude/i);
    const downloadInput = screen.getByPlaceholderText(/download/i);
    const uploadInput = screen.getByPlaceholderText(/upload/i);

    const networksSelect = screen.getByLabelText("Networks Available");
    const distancesSelect = screen.getByLabelText("Distance Classes");
    const technologySelect = screen.getByLabelText("Technology Types");
    const connectionSelect = screen.getByLabelText("Connection Types");
    const contractSelect = screen.getByLabelText("Contract Terms");

    await userEvent.type(latitudeInput, "90.00000");
    await userEvent.type(longitudeInput, "90.00000");
    await userEvent.type(downloadInput, "1000");
    await userEvent.type(uploadInput, "500");

    fireEvent.keyDown(networksSelect, {
      key: "ArrowDown",
      keyCode: 40,
      code: 40,
    });
    const networksSelectItem = screen.getByText(/Latwan/i);
    fireEvent.click(networksSelectItem);

    fireEvent.keyDown(distancesSelect, {
      key: "ArrowDown",
      keyCode: 40,
      code: 40,
    });
    const distancesSelectItem = screen.getByText(/offnetlabel/i);
    fireEvent.click(distancesSelectItem);

    fireEvent.keyDown(technologySelect, {
      key: "ArrowDown",
      keyCode: 40,
      code: 40,
    });
    const technologySelectSelectItem = screen.getByText(/ADSL/i);
    fireEvent.click(technologySelectSelectItem);

    fireEvent.keyDown(connectionSelect, {
      key: "ArrowDown",
      keyCode: 40,
      code: 40,
    });
    const connectionSelectSelectItem = screen.getByText(/DIA/i);
    fireEvent.click(connectionSelectSelectItem);

    fireEvent.keyDown(contractSelect, {
      key: "ArrowDown",
      keyCode: 40,
      code: 40,
    });
    const contractSelectSelectItem = screen.getByText(/12/i);
    fireEvent.click(contractSelectSelectItem);

    expect(mockHandleSelectChange).toHaveBeenCalledTimes(5);
    expect(mockHandleChange).toHaveBeenCalledTimes(23);

    await userEvent.click(button);

    expect(mockHandleSubmit).toHaveBeenCalled();
    expect(dataJSON).not.toBeInTheDocument();
  });

  it("Should match snapshot", () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
