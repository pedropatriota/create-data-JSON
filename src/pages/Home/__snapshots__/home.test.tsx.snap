// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Home page Should match snapshot 1`] = `
<div>
  <div
    class="sc-kAyceB cLudBL"
  >
    <form
      class="sc-imWYAI CGaiv"
    >
      <label
        for="addresses"
      >
        Addresses (lat, long)
        <div
          class="sc-fqkvVR bGSaWp"
        >
          <div
            class="sc-dcJsrY hsPXea"
          >
            <input
              class="sc-iGgWBj csJhxu"
              id="latitude"
              name="latitude"
              placeholder="Latitude"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="sc-fqkvVR bGSaWp"
        >
          <div
            class="sc-dcJsrY hsPXea"
          >
            <input
              class="sc-iGgWBj csJhxu"
              id="longitude"
              name="longitude"
              placeholder="Longitude"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </label>
      <div>
        <label
          for="react-select-2-input"
        >
          Distance Classes
        </label>
        <div
          class="sc-eqUAAy afNuV css-b62m3t-container"
          id="Distance Classes"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-20-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-13cymwt-control"
          >
            <div
              class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                class="react-select__placeholder css-1jqq78o-placeholder"
                id="react-select-20-placeholder"
              >
                Select...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-20-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Distance Classes"
                  aria-required="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-20-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            aria-hidden="true"
            class="css-5kkxb2-requiredInput-RequiredInput"
            required=""
            tabindex="-1"
            value=""
          />
        </div>
      </div>
      <div>
        <label
          for="react-select-2-input"
        >
          Technology Types
        </label>
        <div
          class="sc-eqUAAy afNuV css-b62m3t-container"
          id="Technology Types"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-21-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-13cymwt-control"
          >
            <div
              class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                class="react-select__placeholder css-1jqq78o-placeholder"
                id="react-select-21-placeholder"
              >
                Select...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-21-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Technology Types"
                  aria-required="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-21-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            aria-hidden="true"
            class="css-5kkxb2-requiredInput-RequiredInput"
            required=""
            tabindex="-1"
            value=""
          />
        </div>
      </div>
      <div>
        <label
          for="react-select-2-input"
        >
          Connection Types
        </label>
        <div
          class="sc-eqUAAy afNuV css-b62m3t-container"
          id="Connection Types"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-22-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-13cymwt-control"
          >
            <div
              class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                class="react-select__placeholder css-1jqq78o-placeholder"
                id="react-select-22-placeholder"
              >
                Select...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-22-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Connection Types"
                  aria-required="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-22-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            aria-hidden="true"
            class="css-5kkxb2-requiredInput-RequiredInput"
            required=""
            tabindex="-1"
            value=""
          />
        </div>
      </div>
      <div>
        <label
          for="react-select-2-input"
        >
          Networks Available
        </label>
        <div
          class="sc-eqUAAy afNuV css-b62m3t-container"
          id="Networks Available"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-23-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-13cymwt-control"
          >
            <div
              class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                class="react-select__placeholder css-1jqq78o-placeholder"
                id="react-select-23-placeholder"
              >
                Select...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-23-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Networks Available"
                  aria-required="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-23-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            aria-hidden="true"
            class="css-5kkxb2-requiredInput-RequiredInput"
            required=""
            tabindex="-1"
            value=""
          />
        </div>
      </div>
      <label
        for="bandWidth"
      >
        Bandwidths (Mb)
        <div
          class="sc-fqkvVR bGSaWp"
        >
          <div
            class="sc-dcJsrY hsPXea"
          >
            <input
              class="sc-iGgWBj csJhxu"
              id="download"
              name="download"
              placeholder="Download"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="sc-fqkvVR bGSaWp"
        >
          <div
            class="sc-dcJsrY hsPXea"
          >
            <input
              class="sc-iGgWBj csJhxu"
              id="upload"
              name="upload"
              placeholder="Upload"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </label>
      <div>
        <label
          for="react-select-2-input"
        >
          Contract Terms
        </label>
        <div
          class="sc-eqUAAy afNuV css-b62m3t-container"
          id="Contract Terms"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-24-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-13cymwt-control"
          >
            <div
              class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                class="react-select__placeholder css-1jqq78o-placeholder"
                id="react-select-24-placeholder"
              >
                Select...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-24-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Contract Terms"
                  aria-required="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-24-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            aria-hidden="true"
            class="css-5kkxb2-requiredInput-RequiredInput"
            required=""
            tabindex="-1"
            value=""
          />
        </div>
      </div>
      <div>
        <label
          for="react-select-2-input"
        >
          Area Codes
        </label>
        <div
          class="sc-eqUAAy afNuV css-b62m3t-container"
          id="Area Codes"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-25-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-13cymwt-control"
          >
            <div
              class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                class="react-select__placeholder css-1jqq78o-placeholder"
                id="react-select-25-placeholder"
              >
                Select...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-25-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Area Codes"
                  aria-required="false"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="react-select-25-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-aXZVg kTynDm"
      >
        <button
          class="sc-gEvEer foyIaa"
          type="submit"
        >
          Generate
        </button>
      </div>
    </form>
  </div>
</div>
`;
